### PR TITLE
sphere recognition heuristics and friends

### PR DESCRIPTION
**Pull request by micjoswig**
Carrying over sphere recognition and friends from polymake.  Three-valued logics works now, with recent addition to Polymake.jl.

### CODE REVIEW

File: src/Combinatorics/SimplicialComplexes.jl [hunk 1]
```diff
@@ -399,6 +399,71 @@ function fundamental_group(K::SimplicialComplex)
...
+
+Heuristically check if the abstract simplicial complex `K` is a combinatorial sphere; see [JLLT22](@cite).
+Note that this is undecidable in general.
+Returns true if recognized as a sphere.
+Returns false if not a sphere.
+Returns nothing if heuristics unsuccessful.
+
+# Examples
+```jldoctest
+julia> julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
```
Comment (lgoettgens) [inline src/Combinatorics/SimplicialComplexes.jl]:
```suggestion
julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
```

File: src/Combinatorics/SimplicialComplexes.jl [hunk 2]
```diff
@@ -399,6 +399,71 @@ function fundamental_group(K::SimplicialComplex)
...
+
+Heuristically check if the abstract simplicial complex `K` is a combinatorial ball; see [JLLT22](@cite).
+Note that this is undecidable in general.
+Returns true if recognized as a ball.
+Returns false if not a ball.
+Returns nothing if heuristics unsuccessful.
+
+# Examples
+```jldoctest
+julia> julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
```
Comment (lgoettgens) [inline src/Combinatorics/SimplicialComplexes.jl]:
```suggestion
julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
```

File: src/Combinatorics/SimplicialComplexes.jl [hunk 3]
```diff
@@ -399,6 +399,71 @@ function fundamental_group(K::SimplicialComplex)
...
+
+# Examples
+```jldoctest
+julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
+
+julia> is_sphere(K)
+false
+```
+"""
+is_sphere(K::SimplicialComplex)::Union{Bool,Nothing} = pm_object(K).SPHERE
```
Comment (benlorenz) [inline src/Combinatorics/SimplicialComplexes.jl]:
```suggestion
is_sphere(K::SimplicialComplex) = pm_object(K).SPHERE::Union{Bool,Nothing}
```
The type should be at the end, this way it is just an assertion and does not trigger an automatic conversion.

<details>
<summary>Details</summary>

```julia
julia> is_sphere1(K::SimplicialComplex) = Oscar.pm_object(K).SPHERE::Union{Bool,Nothing}
is_sphere1 (generic function with 1 method)

julia> is_sphere2(K::SimplicialComplex)::Union{Bool,Nothing} = Oscar.pm_object(K).SPHERE
is_sphere2 (generic function with 1 method)

julia> Base.return_types(is_sphere1)
1-element Vector{Any}:
 Union{Nothing, Bool}

julia> Base.return_types(is_sphere2)
1-element Vector{Any}:
 Union{Nothing, Bool}

julia> @code_warntype is_sphere1(torus())
MethodInstance for is_sphere1(::SimplicialComplex)
  from is_sphere1(K::SimplicialComplex) @ Main REPL[17]:1
Arguments
  #self#::Core.Const(is_sphere1)
  K::SimplicialComplex
Body::Union{Nothing, Bool}
1 ─ %1 = Oscar.pm_object::Core.Const(Oscar.pm_object)
│   %2 = (%1)(K)::Polymake.LibPolymake.BigObject
│   %3 = Base.getproperty(%2, :SPHERE)::Any
│   %4 = Core.apply_type(Main.Union, Main.Bool, Main.Nothing)::Core.Const(Union{Nothing, Bool})
│   %5 = Core.typeassert(%3, %4)::Union{Nothing, Bool}
└──      return %5


julia> @code_warntype is_sphere2(torus())
MethodInstance for is_sphere2(::SimplicialComplex)
  from is_sphere2(K::SimplicialComplex) @ Main REPL[18]:1
Arguments
  #self#::Core.Const(is_sphere2)
  K::SimplicialComplex
Locals
  @_3::Any
Body::Union{Nothing, Bool}
1 ─ %1 = Core.apply_type(Main.Union, Main.Bool, Main.Nothing)::Core.Const(Union{Nothing, Bool})
│   %2 = Oscar.pm_object::Core.Const(Oscar.pm_object)
│   %3 = (%2)(K)::Polymake.LibPolymake.BigObject
│   %4 = Base.getproperty(%3, :SPHERE)::Any
│        (@_3 = %4)
│   %6 = (@_3 isa %1)::Bool
└──      goto #3 if not %6
2 ─      goto #4
3 ─ %9 = Base.convert(%1, @_3)::Any
└──      (@_3 = Core.typeassert(%9, %1))
4 ┄      return @_3::Union{Nothing, Bool}

```
</details>

File: src/Combinatorics/SimplicialComplexes.jl [hunk 4]
```diff
@@ -399,6 +399,71 @@ function fundamental_group(K::SimplicialComplex)
...
+
+# Examples
+```jldoctest
+julia> K = simplicial_complex([[1,2,3],[2,3,4]]);
+
+julia> is_ball(K)
+true
+```
+"""
+is_ball(K::SimplicialComplex)::Union{Bool,Nothing} = pm_object(K).BALL
```
Comment (benlorenz) [inline src/Combinatorics/SimplicialComplexes.jl]:
```suggestion
is_ball(K::SimplicialComplex) = pm_object(K).BALL::Union{Bool,Nothing}
```

File: src/Combinatorics/SimplicialComplexes.jl [hunk 5]
```diff
@@ -399,6 +399,71 @@ function fundamental_group(K::SimplicialComplex)
...
+true
+
+julia> is_manifold(simplicial_complex([[1,2],[2,3]]))
+true
+
+julia> is_manifold(simplicial_complex([[1,2],[2,3],[2,4]]))
+false
+```
+"""
+is_manifold(K::SimplicialComplex)::Union{Bool,Nothing} = pm_object(K).MANIFOLD
```
Comment (benlorenz) [inline src/Combinatorics/SimplicialComplexes.jl]:
```suggestion
is_manifold(K::SimplicialComplex) = pm_object(K).MANIFOLD::Union{Bool,Nothing}
```

**Review by benlorenz — APPROVED** [review]

**Pull request merged**